Monads/Option.cs

```diff
@@ -133,4 +133,138 @@ public class Some<T>(T value) : Option<T>(value) where T : class;
     /// </summary>
     /// <typeparam name="T">the type of the value that is missing</typeparam>
     public class None<T>() : Option<T>() where T : class;
+
+    /// <summary>
+    /// The monad type used to represent that there might be a value, but doesn't have to be as well.
+    /// </summary>
+    /// <remarks>Also provides simple shorthands for ways of manipulating with them.</remarks>
+    /// <typeparam name="T">the underlying value</typeparam>
+    public class ValueOption<T> : IEquatable<ValueOption<T>>
+        where T : struct
+    {
+        protected T? _value;
+
+        /// <summary>
+        /// Returns the underlying value
+        /// </summary>
+        /// <returns>the contained value</returns>
+        /// <exception cref="InvalidOperationException">there is no underlying value</exception>
+        public T Value => _value ?? throw new InvalidOperationException();
+
+        /// <summary>
+        /// Returns a value indicating whether the underlying value or is <see langword="null"/> or not.
+        /// </summary>
+        /// <returns>
+        /// <see langword="true"/> if the underlying value exists.
+        /// <see langword="false"/> if there is no underlying value.
+        /// </returns>
+        //[System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]
+        [System.Diagnostics.CodeAnalysis.MemberNotNullWhen(true, nameof(_value))]
+        public bool HasValue => _value is not null;
```

Comment:
Member has to be checked for `null` by `_value.HasValue` instead

```suggestion
        public bool HasValue => _value.HasValue;
```